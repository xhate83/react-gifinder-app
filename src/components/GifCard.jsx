import PropTypes from 'prop-types';

export const GifCard = ({title, url}) => {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifCard.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}


