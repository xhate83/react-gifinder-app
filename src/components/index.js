export * from './AddCategory';
export * from './GifGrid';
export * from './GifCard';