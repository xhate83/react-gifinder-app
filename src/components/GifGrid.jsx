import PropTypes from 'prop-types';
import { GifCard } from './GifCard';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({ category }) => {

    const { gifs, isLoading } = useFetchGifs(category)

    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && ( <h2>Cargando...</h2> )
            }
            <div className='card-grid'>{
                gifs.map( (gif) => (
                   <GifCard key={gif.id} {...gif} />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
