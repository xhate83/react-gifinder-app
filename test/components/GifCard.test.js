import { GifCard } from '../../src/components/GifCard';
import { render, screen } from "@testing-library/react";

describe('GifCard test', () => {
    const title = 'Goku';
    const url = 'https://dragonball.com/imagendegoku.jpg';
    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifCard title={title} url={url}/> );
        expect( container ).toMatchSnapshot()
    });

    test('Debe mostrar el url y el atl indicado', () => {
        render( <GifCard title={title} url={url}/> );
        // expect( screen.getByRole('img').src ).toBe(url);
        // expect( screen.getByRole('img').alt ).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => { 
        render( <GifCard title={title} url={url}/> );
        expect( screen.getByText(title) ).toBeTruthy();
    });
 })