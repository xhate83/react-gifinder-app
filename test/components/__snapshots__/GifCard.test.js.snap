// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard test Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://dragonball.com/imagendegoku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
